collectors/python.d.plugin/nvidia_smi/nvidia_smi.chart.py

```diff
@@ -306,10 +323,15 @@ def processes(self):
         p_nodes = self.root.find('processes').findall('process_info')
         ps = []
         for p in p_nodes:
+            pid = int(p.find('pid').text)
+            proc_stat_file = os.stat("/proc/%d" % pid)
+            uid = proc_stat_file.st_uid
+            username = pwd.getpwuid(uid)[0]
```

Comment:
> proc_stat_file = os.stat("/proc/%d" % pid)
            
This wont work in docker. There is `NETDATA_HOST_PREFIX` env var, if it is set we need `os.path.join(host, 'proc', pid)`.

> username = pwd.getpwuid(uid)[0]

This `pwd.getpwuid` wont work in docker, i think it looks in `/etc/password` and there is no way to set prefix. it is likely we have there `KeyError`.

Reply:
yeah, it seems people use it `nvidia-smi` in docker

```cmd
0 ~ $ docker search netdata-glibc --no-trunc
NAME                       DESCRIPTION                                            STARS               OFFICIAL            AUTOMATED
d34dc3n73r/netdata-glibc   netdata built with glibc for use with nvidia-docker2   2                                       [OK]
```

Reply:
So there is 2 options:
 - do not try to collect user mem usage when inside a container
 - write our own `pwd.getpwuid` that can use `NETDATA_HOST_PREFIX`. By default we mount `/etc/passwd` to `/host/etc/passwd`, it is needed by `apps.plugin`

There is python (py2, old code) code we could use, it is python implementation of `pwd` package

https://github.com/enthought/Python-2.7.3/blob/master/Lib/plat-os2emx/pwd.py

Reply:
I tried to use the code you linked;
I cannot test in a docker right now...